app/models/scheme.rb

```diff
@@ -295,6 +295,16 @@ def active?
     status == :active
   end
 
+  def has_active_locations?
+    active? && locations.active.exists?
+  end
+
+  def has_active_locations_on_date?(date)
+    return false unless date
+
+    locations.any? { |location| date && location.active_on_date?(date) }
```

Comment:
I'd maybe like to use scopes here too. Especially if an org has many locations.
Maybe worth refactoring `active` scope (and all the relevant statuses scopes) to take in a conditional date and have `now` by default (I assume that's possible)


You could then call it here something like this: `locations.active_status(date).any?`

Reply:
Ok will do